src/openms/source/ANALYSIS/OPENSWATH/TransitionTSVFile.cpp

```diff
@@ -227,8 +227,25 @@ namespace OpenMS
     }
 
     bool spectrast_legacy = false; // we will check below if SpectraST was run in legacy (<5.0) mode or if the RT normalization was forgotten.
-    int cnt = 0;
-    while (TextFile::getLine(data, line)) // make sure line endings are handled correctly
+    size_t cnt = 0;
+    size_t endline;
+
+    if (batchSize <= 0)
+    {
+      endline = std::numeric_limits<int>::max();
+    }
+    else
+    {
+      endline = offset + batchSize;
+      // stream the offsets to null
+      for (int i = 0; i < offset; ++i)
+      {
+        std::getline(data, line);
+        cnt++;
+      }
```

Comment:
Do you sweeps through the file multiple times to find the start in line offset? Is this necessary? It might be possible to store the current file pointer position and then use file.seekg(number_of_bytes); to continue reading where one left off - without rereading all lines until offset from disc.